### DISCUSSION TITLE
Update on Checking Status for NASA AppEEARS Data Requests

### DISCUSSION THREAD

**Discussion by anatarano** [cu-esiil-edu/2024-data-short-course#78] [General]
In our second coding challenge, [Vegetation](https://cu-esiil-edu.github.io/esiil-learning-portal/shortcourse/notebooks/05-vegetation/vegetation.html), we request MODIS data using an API, which is authenticated by our NASA Earthdata account information. During yesterday's lecture, @eculler walked me through a few ways to check on the status of my data requests. I just wanted to share some updates here since it might be helpful for others. 

### 💌 Email Notifications from "AppEEARS"
This is by far the most convenient way to determine the status of your data request. You first receive an email titled "**AppEEARS cp-ndvi Request Received**" from _appeears-noreply@nasa.gov_. Once your request is fulfilled, you receive another email titled "**AppEEARS cp-ndvi Request Complete**" from the same email address. 

⚠️ Note that if you change the `download_key` value from `'cp-ndvi'` when calling the downloader, your email titles may be different. See below for code sample.

```python
ndvi_downloader = eaapp.AppeearsDownloader(
    download_key='cp-ndvi',
    ea_dir=data_dir,
    product='MOD13Q1.061',
    layer='_250m_16_days_NDVI',
    start_date="07-01",
    end_date="07-31",
    recurring=True,
    year_range=[2018, 2023],
    polygon=cameron_peak_gdf
)
``` 
### 📑 Logs in Code
Yesterday in lecture, @eculler walked me through generating logs using the following code before calling the downloader.
```python
import logging # Logging library import
logging.basicConfig(level=logging.DEBUG) # log redirection
``` 
During the lecture, the logs didn't have an output, which was surprising and weird. I am still not sure of what happened and if I will ever be able to reproduce what happened in lecture. But when I re-ran that cell today, the logs were very descriptive and showed that my status is pending, queued, and complete (see screenshot below). This is also super helpful for live information.

<img width="1440" alt="Screen Shot 2024-04-30 at 11 12 45 AM" src="https://github.com/cu-esiil-edu/2024-data-short-course/assets/10750991/d31a5a99-492c-44ef-a5c6-da77bae2252a">

 ### 🕸 AppEEARS Website
@eculler also showed me how to access the different requests using the[ AppEEARS Explore page](https://appeears.earthdatacloud.nasa.gov/explore). There you can also see what data you requested, when, and its status. See screenshot below.
<img width="1440" alt="Screen Shot 2024-04-30 at 11 28 18 AM" src="https://github.com/cu-esiil-edu/2024-data-short-course/assets/10750991/f4683a03-25b5-452d-8255-a46b1228bdf8">

**Comment by nquarder** [comment]
Thank you @anatarano this is a very helpful resource!
